src/utils/index.ts

```diff
@@ -216,3 +217,37 @@ export function shortenText(text: string, length = 20, separator = '[…]'): str
 
   return `${text.slice(0, length)}${separator}${text.slice(-length)}`
 }
+
+const DEFAULT_POLLING_FREQUENCY = 1_000
+const DEFAULT_STAMP_USABLE_TIMEOUT = 120_000
+
+interface Options {
+  pollingFrequency?: number
+  timeout?: number
+}
+
+export function waitUntilStampUsable(batchId: BatchId, beeDebug: BeeDebug, options?: Options): Promise<PostageBatch> {
+  const timeout = options?.timeout || DEFAULT_STAMP_USABLE_TIMEOUT
+  const pollingFrequency = options?.pollingFrequency || DEFAULT_POLLING_FREQUENCY
+  let timeoutReached = false
+
+  const timeoutPromise = (): Promise<never> =>
+    new Promise((_, reject) =>
+      setTimeout(() => {
+        timeoutReached = true
+        reject(new Error('Wait until stamp usable timeout has been reached'))
+      }, timeout),
+    )
+
+  const stampWaitPromise = async (): Promise<PostageBatch | undefined> => {
+    while (!timeoutReached) {
+      const stamp = await beeDebug.getPostageBatch(batchId)
+
+      if (stamp.usable) return stamp
+      await sleepMs(pollingFrequency)
+    }
+  }
```

Comment:
I usually do this with for loops, I think the upside is you don't need to race promises, no need for a separate setTimeout for the rejection, and the implementation is a single block that is simpler IMHO.

I am fine with this as well, just wondering if this has some pros I am unaware of 🙂

Reply:
Hey! Actually no benefits, mostly really just complications. I will refactor, thanks for the suggestion :)

Reply:
The only advantage is that it spends exactly the `timeout` time unlike for loop which also adds the time spent on requests to check the stamp is usable . I don't see that as a problem, just wanted to point it out :)